x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_exploration/components/regression_exploration/results_table.tsx

```diff
@@ -0,0 +1,482 @@
+                  }
+                )}
+              >
+                <EuiBadge>
+                  {i18n.translate(
+                    'xpack.ml.dataframe.analytics.regressionExploration.indexObjectBadgeContent',
+                    {
+                      defaultMessage: 'object',
+                    }
+                  )}
+                </EuiBadge>
+              </EuiToolTip>
+            );
+          }
+
+          return d;
+        };
+
+        let columnType;
+
+        if (tableItems.length > 0) {
+          columnType = typeof tableItems[0][k];
+        }
+
+        if (typeof columnType !== 'undefined') {
+          switch (columnType) {
+            case 'boolean':
+              column.dataType = 'boolean';
+              break;
+            case 'Date':
+              column.align = 'right';
+              column.render = (d: any) =>
+                formatHumanReadableDateTimeSeconds(moment(d).unix() * 1000);
+              break;
+            case 'number':
+              column.dataType = 'number';
+              column.render = render;
+              break;
+            default:
+              column.render = render;
+              break;
+          }
+        } else {
+          column.render = render;
+        }
+
+        return column;
+      })
+    );
+  }
+
+  useEffect(() => {
+    if (jobConfig !== undefined) {
+      const predictedFieldName = getPredictedFieldName(
+        jobConfig.dest.results_field,
+        jobConfig.analysis
+      );
+      const predictedFieldSelected = selectedFields.includes(predictedFieldName);
+
+      const field = predictedFieldSelected ? predictedFieldName : selectedFields[0];
+      const direction = predictedFieldSelected ? SORT_DIRECTION.DESC : SORT_DIRECTION.ASC;
+      loadExploreData({ field, direction, searchQuery });
+      return;
+    }
+  }, [JSON.stringify(searchQuery)]);
+
+  useEffect(() => {
+    // by default set the sorting to descending on the prediction field (`<dependent_varible or prediction_field_name>_prediction`).
+    // if that's not available sort ascending on the first column.
+    // also check if the current sorting field is still available.
+    if (jobConfig !== undefined && columns.length > 0 && !selectedFields.includes(sortField)) {
+      const predictedFieldName = getPredictedFieldName(
+        jobConfig.dest.results_field,
+        jobConfig.analysis
+      );
+      const predictedFieldSelected = selectedFields.includes(predictedFieldName);
+
+      const field = predictedFieldSelected ? predictedFieldName : selectedFields[0];
+      const direction = predictedFieldSelected ? SORT_DIRECTION.DESC : SORT_DIRECTION.ASC;
+      loadExploreData({ field, direction, searchQuery });
+      return;
+    }
+  }, [jobConfig, columns.length, sortField, sortDirection, tableItems.length]);
+
+  let sorting: SortingPropType = false;
+  let onTableChange;
+
+  if (columns.length > 0 && sortField !== '' && sortField !== undefined) {
```

Comment:
I am seeing a strange issue in this example. I enter a search of `p3<-1`, and table renders as expected:
![image](https://user-images.githubusercontent.com/7405507/67942351-8f26d680-fbcf-11e9-99d4-10469558a9aa.png)

But then when I try to sort on the `p3` column, I get no results. Possibly related to it being the prediction field? I only seem to get this when sorting on the field used for prediction:
![image](https://user-images.githubusercontent.com/7405507/67942392-a06fe300-fbcf-11e9-9d93-9f0e9365ab82.png)

Reply:
The outlier results view might have similar issues, as it’s based on the same code. Along with the other suggested changes that impact both views, I will address that in follow-ups.